Source/Runtime/EmscriptenIntrinsics.cpp

```diff
@@ -224,6 +224,17 @@ namespace Runtime
 	{
 		return ungetc(character,vmFile(file));
 	}
+	DEFINE_INTRINSIC_FUNCTION4(env,_fread,_fread,I32,I32,pointer,I32,size,I32,count,I32,file)
+	{
+		if(pointer + uint64(size) * (count + 1) > instanceAddressSpaceMaxBytes)
```

Comment:
You can only pass in 2^32-1, so:
`pointer + size * (uint64(count) + 1) = 0xFFFFFFFF + 0xFFFFFFFF * (0xFFFFFFFF + 1) = 0xFFFFFFFF + 0xFFFFFFFF00000000 = 0xFFFFFFFFFFFFFFFF`; no overflow